drivers/pinctrl/pinctrl_rpi_pico.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2021 Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/pinctrl.h>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+#include <hardware/gpio.h>
+#pragma GCC diagnostic pop
```

Comment:
can you explain why is this required? I can't see anything suspicious in https://github.com/yonsch/hal_rp_demo/blob/master/pico-sdk/src/rp2_common/hardware_gpio/include/hardware/gpio.h that would justify this

Reply:
Some header files high in the dependency tree define static, always-inline functions. These are not used by every source file, but are included anyway, causing many unused-function warnings